### ISSUE TITLE
variable with name `await` is rejected 

### ISSUE THREAD

**Issue by AYOZEMONZON** [squaremo/amqp.node#251]
Using amqplib in an ionic/typescript context [here](https://github.com/AYOZEMONZON/SnappyMap/tree/add-rabbitmq-1) we have (among other errors) the issue that `await` is used as variable name 

```javascript
ConfirmChannel.prototype.waitForConfirms = function(k) {
  var await = [];
  var unconfirmed = this.unconfirmed;
  unconfirmed.forEach(function(val, index) {
    if (val === null); // already confirmed
    else {
      var confirmed = defer();
      unconfirmed[index] = function(err) {
        if (val) val(err);
        if (err === null) confirmed.resolve();
        else confirmed.reject(err);
      };
      await.push(confirmed.promise);
    }
  });
  return when.all(await).then(function() { k(); },
                              function(err) { k(err); });
};
```

whcih is rejected with error message 
```
JavaScript critical error at line 914, column 7 in ms-appx-web://com.ionicframework.snappymappy728356/www/build/js/app.bundle.js\n\nSCRIPT1048: The use of a keyword for an identifier is invalid
```

How can we circumvent this problem?

**Comment by dharness** [comment]
You really shouldn't be using await a variable name since it is a keyword.
`https://github.com/tc39/ecmascript-asyncawait`

**Comment by AntonioMendez** [comment]
@dharness the release version from npm still uses it. 

![image](https://cloud.githubusercontent.com/assets/10880610/15495106/63b1dfd2-218e-11e6-8371-cff2b4b6580d.png)

**Comment by dimofte** [comment]
The issue has been solved in 526124d, but it hadn't got its own release yet. 

You can use this forked [pre-release](https://github.com/ShopCo/amqp.node/releases/tag/v0.4.1.1), but only as a temporary solution, because I'll remove the fork as soon as @squaremo makes the next release. It's the current version (v.0.4.1) with the commit added cherrypick. 
If you use it, be sure to run `make` before using it (e.g if you use it with npm in a project, add a script to package.json, `"postinstall": "cd node_modules/amqplib && make"`)

**Comment by leebenson** [comment]
can this be closed with the release of [v0.4.2](https://github.com/squaremo/amqp.node/releases/tag/v0.4.2)?

**Comment by squaremo** [comment]
In the absence of complaints about v0.4.2, yes I think so :)
